### PR TITLE
Update Docs to include info about how to reuse weights

### LINKED ISSUE #1084: [Feature Request] Documentation around multiple pipelines reusing model weights
**Is your feature request related to a problem? Please describe.**
If I have a StableDiffusionPipeline, a StableDiffusionImg2ImgPipeline, and a StableDiffusionInpaintPipeline, these will all load separate weights to the GPU even though they are all using the same model. This makes creating applications that use the full breadth of SD difficult to create.

**Describe the solution you'd like**
It seems like it _is_ possible to reuse weights, according to https://discuss.huggingface.co/t/access-clip-from-stablediffusionpipeline-and-use-the-same-models-for-multiple-pipelines/24813. However, this is not at all clear from the API docs and it's also unclear whether we need to do things like route to cuda multiple times.

**Describe alternatives you've considered**
NA

**Additional context**
NA

### PR DESCRIPTION
**Pull request by theahura**
Resolves #1084

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -330,6 +330,32 @@ prompt = "Face of a yellow cat, high resolution, sitting on a park bench"
...
 ```
 
+### Weight Reuse
+
+It's possible to reuse weights between pipelines. For example, if you wanted to use a StableDiffusionPipeline and a StableDiffusionImg2ImgPipeline, you could initialize the weights for the first and apply them to the latter:
+```python
+# make sure you're logged in with `huggingface-cli login`
+from diffusers import StableDiffusionPipeline, StableDiffusionImg2ImgPipeline
+
+SD = StableDiffusionPipeline.from_pretrained("runwayml/stable-diffusion-v1-5")
```
Comment (pcuenca) [inline README.md:340]: nit: all uppercase var names don't match the style of the rest of the code or READMEs, in my opinion. Would it be possible to find some lower case names instead? :)

**Review by pcuenca — APPROVED** [review]
Thanks for this, it's very useful!

I think we could potentially take the opportunity to explain the use of `components`, perhaps in the context of using different schedulers. This is something that comes up once in a while, and could be an additional example to give here.

This would be a code snippet that explains what I'm referring to:

```Python
import torch
from diffusers import StableDiffusionPipeline, EulerAncestralDiscreteScheduler

model_id = "runwayml/stable-diffusion-v1-5"
pipe = StableDiffusionPipeline.from_pretrained(model_id)

euler = EulerAncestralDiscreteScheduler.from_config(model_id, subfolder="scheduler")
components = pipe.components
components["scheduler"] = euler
euler_pipe = StableDiffusionPipeline(**components)
```

Do you think it makes sense to add such an example here?

File: README.md (hunk 1 above)
Comment (theahura) [reply README.md]: done

**Comment by theahura** [comment]
> I think we could potentially take the opportunity to explain the use of components, perhaps in the context of using different schedulers.

I don't think this subsection is the right place for it, and imo that is probably going too in depth for the average user of this library (components are meant to be abstracted away by the pipelines so introducing them in the README is probably extraneous). 

I'm also not the right person to write that section :sweat_smile:

**Comment by pcuenca** [comment]
Cool, that's fair :)

File: README.md [hunk 2]
```diff
@@ -330,6 +330,32 @@ prompt = "Face of a yellow cat, high resolution, sitting on a park bench"
...
+
+It's possible to reuse weights between pipelines. For example, if you wanted to use a StableDiffusionPipeline and a StableDiffusionImg2ImgPipeline, you could initialize the weights for the first and apply them to the latter:
+```python
+# make sure you're logged in with `huggingface-cli login`
+from diffusers import StableDiffusionPipeline, StableDiffusionImg2ImgPipeline
+
+sd = StableDiffusionPipeline.from_pretrained("runwayml/stable-diffusion-v1-5")
+
+# Note that each subcomponent can be added or removed separately.
+sd_img2img = StableDiffusionImg2ImgPipeline.from_pretrained(
```
Comment (patrickvonplaten) [inline README.md:343]:
```suggestion
sd_img2img = StableDiffusionImg2ImgPipeline.from_pretrained(**sd_img2img.components)
```
Should do the same in just one line -> should we  maybe advertise this instead? :-) 
See https://huggingface.co/docs/diffusers/api/diffusion_pipeline#diffusers.DiffusionPipeline.components

**Pull request closed without merging**